pkg/tsdb/Magefile.go

```diff
@@ -0,0 +1,112 @@
+//go:build mage
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	build "github.com/grafana/grafana-plugin-sdk-go/build"
+)
+
+func find(dir string, name string) ([]string, error) {
+	files := []string{}
+	err := filepath.Walk(dir, func(path string, f os.FileInfo, err error) error {
+		if filepath.Base(path) == name {
+			files = append(files, path)
+		}
+		return nil
+	})
+
+	return files, err
+}
+
+func findPluginJSONDir(pluginDir string) (string, error) {
+	pluginJSONMatches, err := find(filepath.Join("../../public/app/plugins/datasource", pluginDir), "plugin.json")
+	if err != nil {
+		return "", err
+	}
+	if len(pluginJSONMatches) == 0 {
+		return "", fmt.Errorf("Could not find plugin.json")
+	}
+	pluginJSONPath := ""
+	for _, pluginJSONMatch := range pluginJSONMatches {
+		// Ignore dist folder
+		if !strings.Contains(pluginJSONMatch, "dist") {
```

Comment:
Nit: Maybe it's more robust to check the folder that contains `plugin.json`, and checking that that part is `dist` instead?

```suggestion
		if filepath.Dir(pluginJSONMatch) != "dist" {
```